### ISSUE TITLE
23.x theme introduces element selectors

### ISSUE THREAD

**Issue by kfeinUI** [ag-grid/ag-grid#3737]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Requesting Community Support**
Community member raising an issue to your attention.

**Providing a Reproducible Scenario**
Found that 23.0.0 introduced element selectors, which is a bad practice that may clobber styling of custom elements due to greater specificity. I'm using the Material theme. Haven't checked the others, but I would imagine there's a good chance that they're following the same pattern.

Embed any sort of custom HTML in the grid. It may be a button, input, or textarea (perhaps others). Do this, for example, within a custom cell renderer. The selectors are just on the grid container, so where inside doesn't really matter.

I'll provide a CodePen if you really need it, but this is very easy to reproduce on the fly. Basic styling scenario. 

**Current behavior**

These rules with element selectors in the theme now have greater specificity than custom element with a CSS module class applied.

**Expected behavior**

Don't use element selectors. They lack flexibility. Target the out of the box grid code that you wish to style using class selectors. If someone truly wishes to apply that styling to their custom creation, they may tap into the class.

**Please tell us about your environment:**

Doesn't matter.

* **ag-Grid version:** 23.0.0

* **Browser:**

Doesn't matter.

* **Language:** all

**Comment by rmc-software** [comment]
Thanks for raising this with us. This has now been fixed in our latest patch v23.0.2.

**Comment by kfeinUI** [comment]
CSS output looks better, though I suggest you give it another pass. There's still some remaining. Think there's a button rule and some targeting the range flavor of input.

**Comment by rmc-software** [comment]
Sorry about that - we'll try and get them all next time. We have created an internal jira to track this: AG-4018.

See our pipeline for progress updates.

https://www.ag-grid.com/ag-grid-pipeline/

A fix is scheduled for our next release v23.1 due in 4/5 weeks.
